src/components/MavenVersionPicker.js

```diff
@@ -14,7 +14,7 @@ export default class MavenVersionPicker extends Component {
 
   constructor(props) {
     super(props)
-    this.state = { customValues: [], options: [] }
+    this.state = { customValues: [], options: [], selected: props.request.revision ? [props.request.revision] : [] }
```

Comment:
All of the changes here and in the other pickers are duplicate. Need some way to either eliminate the need for the code or share it.

Related, it feels like this scenario is not that unusual and ought not be this hard to implement. This approach feels sort of subtle. I don't know a better one right now but if feels overly complicated. (could just be my naivety)